pkg/epp/scheduling/framework/plugins.go

```diff
@@ -24,18 +24,29 @@ import (
 )
 
 const (
-	ProfilePickerType   = "ProfilePicker"
-	FilterPluginType    = "Filter"
-	ScorerPluginType    = "Scorer"
-	PickerPluginType    = "Picker"
-	PostCyclePluginType = "PostCycle"
+	ProfilePickerType          = "ProfilePicker"
+	FilterPluginType           = "Filter"
+	ScorerPluginType           = "Scorer"
+	PickerPluginType           = "Picker"
+	PostCyclePluginType        = "PostCycle"
+	ProcessProfilesResultsType = "ProcessProfilesResults"
 )
 
-// ProfilePicker selects the SchedulingProfiles to run from a list of candidate profiles, while taking into consideration the request properties
-// and the previously executed SchedluderProfile cycles along with their results.
-type ProfilePicker interface {
+// ProfileHandler defines the extension points for handling multi SchedulerProfile instances.
+// More specifically, this interface defines the 'PickProfiles' and 'ProcessProfilesResults'
+// extension points.
+
+// ProfileHandler
+type ProfileHandler interface {
 	plugins.Plugin
-	Pick(ctx context.Context, request *types.LLMRequest, profiles map[string]*SchedulerProfile, executionResults map[string]*types.Result) map[string]*SchedulerProfile
+	// Pick selects the SchedulingProfiles to run from a list of candidate profiles, while taking into consideration the request properties
+	// and the previously executed SchedluderProfile cycles along with their results.
+	Pick(ctx context.Context, request *types.LLMRequest, profiles map[string]*SchedulerProfile, profileResults map[string]*types.ProfileRunResult) map[string]*SchedulerProfile
```

Comment:
Perhaps we need a different name so we don't confuse it with the endpoint picker plugin. One suggestion is `Select`.

Reply:
Generally speaking, I'd like to be consistent in terminology.
if we use `Pick` function in the `Picker` interface, keeping consistency here means having `Pick` function in the `ProfileHandler` interface.
I'm ok with changing it to `Select` but in such a case I'd consider changing in all places to keep it consistent - 
e.g., rename `Picker` to `Selector` and rename `Pick` function to `Select` in both places.

LMKWYT.